### ISSUE TITLE
Instant vCPU shutdown request on Windows with qemu: mmio_fetch_instruction() failed

### ISSUE THREAD

**Issue by grische** [intel/haxm#211]
**Describe the Bug**

Summary: 
I am trying to build a Debian 9.8 image with packer on Windows (details see [#Host Environment]) with `accelerator=hax` and when I assign the qemu VM `1024MB`, it shows the screen for the VM for less than 500ms and crashes.
If I assign `1536MB` or more, it does not crash and works as expected.
If I use no accelerator, it does not crash even with `512MB`.

**Host Environment**

* HAXM version: v7.5.1
* Host OS version: Windows 10 Pro version 1809
* Host OS architecture: x86\_64
* Host CPU model: Intel Core i5-7200
* Host RAM size: 16 GB
* (Optional) Host computer model: Lenovo T470

**Guest Environment**

* Android Emulator or QEMU version: QEMU emulator version 3.1.50 (v3.1.0-13607-geb2db0f7ba-dirty)
* Guest OS version: Crashes before guest OS is booted
* Guest OS architecture: amd64

**To Reproduce**

Start qemu and see how it instantly crashes

Here are the respective lines of PACKER_LOG=1
```
2019/06/12 14:59:00 packer.exe: 2019/06/12 14:59:00 Executing C:\Program Files\qemu\qemu-system-x86_64.exe: []string{"-machine", "type=pc,accel=hax", "-device", "virtio-net,netdev=
user.0", "-boot", "once=d", "-smp", "cpus=2,sockets=2", "-cdrom", "C:\\Users\\user\\Documents\\image-pipeline\\debian\\packer_cache\\663ed9180cb52b92ef1f3c57cf2fb11e7
66685b6.iso", "-m", "1024M", "-netdev", "user,id=user.0,hostfwd=tcp::2981-:22", "-display", "sdl", "-vnc", "127.0.0.1:14", "-name", "debian-9.8-image", "-drive", "file=..\\bui
lds\\packer-debian-9.8-image-debian-9.8-amd64-qemu-2019-06-12T145859\\debian-9.8-image,if=virtio,cache=writeback,discard=ignore,format=qcow2"}
2019/06/12 14:59:00 packer.exe: 2019/06/12 14:59:00 Started Qemu. Pid: 10940
2019/06/12 14:59:00 packer.exe: 2019/06/12 14:59:00 Qemu stdout: HAX is working and emulator runs in fast virt mode.
2019/06/12 14:59:01 packer.exe: 2019/06/12 14:59:01 Qemu stdout: VCPU shutdown request
2019/06/12 14:59:01 packer.exe: 2019/06/12 14:59:01 Qemu stdout: VCPU shutdown request
```

**Reproducibility**

It has a high chance of reproducibility, but very randomly it does not crash. It might be related to system load.

**Diagnostic Information**

Excerpt from the HAXM log:
```
00179234	8.73660755	haxm_error:gpa_space_map_range: start_gpa=0x40000000 is reserved for MMIO	
00179235	8.73661232	haxm_error:gpa_space_map_page: gpa_space_map_range() returned -22	
00179236	8.73661613	haxm_error:mmio_map_guest_virtual_page_slow: gpa_space_map_page() failed: vcpu_id=1, gva=0x40000b00, gpa=0x40000000	
00179237	8.73661995	haxm_panic:vcpu_emulate_insn: mmio_fetch_instruction() failed: vcpu_id=1, gva=0x40000b00 (CS:IP=0x0:0x40000b00)	
```

Full HAXM log:
[haxm_crash.log](https://github.com/intel/haxm/files/3282090/haxm_log_crash.log)

**Additional context**

None

**Comment by HaHoYou** [comment]
See #209 and #205, since you are using qemu 3.1, and cpus=2, you may want to try low version of qemu 3.0.0.

**Comment by grische** [comment]
@HaHoYou downgrading to Qemu 3.0 fixed the problem, but neither #209 nor #205 mention anything about memory.

All in all, they might be a duplicate of #171, but I am waiting for confirmation from @krytarowski.

**Comment by HaHoYou** [comment]
The fix patch has been upstreamed to QEMU: 6f38dca615286796df9a967117f3ce42b918c8db. FYI, if you need higher version of qemu.

**Comment by josephernest** [comment]
@HaHoYou I think I have the still issue ; I tried with QEMU 3.0.0, 3.1.0, 4.2.0, and I have the same problem with all three.

See https://github.com/intel/haxm/issues/307
